api/app/src/providers/withings.ts

```diff
@@ -247,24 +249,66 @@ export class Withings extends Provider implements OAuth2 {
     return mapToActivity(date, activity, workouts);
   }
 
-  async getBiometricsData(connectedUser: ConnectedUser, date: string): Promise<Biometrics> {
+  async fetchBodyData(accessToken: string, date: string) {
     const params = {
-      action: "list",
+      action: "getmeas",
       startdate: dayjs(date).unix(),
+      enddate: dayjs(date).add(1, "day").unix(),
     };
 
+    const response = await axios.post(`${Withings.URL}/measure`, null, {
+      headers: {
+        Authorization: `Bearer ${accessToken}`,
+      },
+      params,
+    });
+
+    return response.data;
+  }
+
+  async getBodyData(connectedUser: ConnectedUser, date: string): Promise<Biometrics> {
     const accessToken = await this.getAccessToken(connectedUser);
 
-    return this.oauth.fetchProviderData<Biometrics>(
+    const response = await this.fetchBodyData(accessToken, date);
+
+    console.log(response.body);
+
+    return mapToBody(date, withingsMeasurementResp.parse(response.body));
+  }
+
+  async fetchHeartData(accessToken: string, date: string): Promise<WithingsHeartRate> {
+    const params = {
+      action: "list",
+      startdate: dayjs(date).unix(),
+    };
+
+    return this.oauth.fetchProviderData<WithingsHeartRate>(
       `${Withings.URL}/${Withings.API_PATH}/heart`,
       accessToken,
       async resp => {
-        return mapToBiometrics(date, withingsHeartRateResp.parse(resp.data.body.series));
+        return withingsHeartRateResp.parse(resp.data.body.series);
       },
       params
     );
   }
 
+  async getBiometricsData(connectedUser: ConnectedUser, date: string): Promise<Biometrics> {
+    const accessToken = await this.getAccessToken(connectedUser);
+
+    const [resHeart, resBody] = await Promise.allSettled([
+      this.fetchHeartData(accessToken, date),
+      this.fetchBodyData(accessToken, date),
+    ]);
+
+    const heart = resHeart.status === "fulfilled" ? resHeart.value : undefined;
+    const body = resBody.status === "fulfilled" ? resBody.value : undefined;
+
+    if (!heart && !body) {
+      throw new Error("All Requests failed");
+    }
+    return mapToBiometrics(date, heart, body.body);
```

Comment:
`body` might be undefined here.

Weird `lint-staged` allowed this to pass... it should've caught it. Or did you bypass it?